### PR TITLE
D3 show full date range user defined in chart & fix chart not showing chart title

### PR DESCRIPTION
**Pull request by prestonfff**
- show full date range user defined in chart
- fix chart not showing chart title

### CODE REVIEW

File: ArgusWeb/app/js/directives/charts/chart.js [hunk 1]
```diff
@@ -15,7 +15,7 @@ function(Metrics, Annotations, ChartRenderingService, ChartDataProcessingService
         // assign items to new $scope
         lineChartScope.chartConfig = {
             chartId: newChartId,
-            chartTitle: scope.options.title || ''   // title is an attribute of <option>, and could be undefined
+            chartTitle: scope.options['title.text'] || ''   // title is an attribute of <option>, and could be undefined
```
Comment (cnardi-dev) [inline ArgusWeb/app/js/directives/charts/chart.js]: cool! empty is ok

Comment (xizi-xu) [inline ArgusWeb/app/js/directives/charts/chart.js]: We still do not support title style at this point.

**Comment by prestonfff** [comment]
@xizi-xu That will involve some metric which does not return whole period of data. I test it locally since the local will not have a full 30days data for example. Will be a better idea to find such metrics in production.

**Comment by xizi-xu** [comment]
bug
Did this happen to you? @prestonfff

**Comment by prestonfff** [comment]
@xizi-xu  just see that. I'm debugging it.

**Comment by prestonfff** [comment]
@xizi-xu Sorry I saw that in my current code. The PR should be fine.

**Comment by cnardi-dev** [comment]
We can check with Tom for metrics to help with testing this fully.  I'll send a note to him shortly.

**Comment by prestonfff** [comment]
@xizi-xu Sorry, I actually saw that because I am on the current code. The PR seems fine, but I do notice some random issue. We can recreate the bug tomorrow and see what is wrong.

File: ArgusWeb/app/js/directives/charts/chart.js (hunk 1 above)
Comment (cnardi-dev) [inline ArgusWeb/app/js/directives/charts/chart.js]: I will be adding in support for title.style

**Comment by xizi-xu** [comment]
I don't understand why this PR got merged. There are known issues. We should try to keep our repo as bug free as possible.

**Comment by prestonfff** [comment]
@xizi-xu For the bug on the gif, I think it is fixed in the next PR.

**Pull request merged**